Fixing and updating the docs

Davi-the-Mudkip opened the pull request:
Some people complained about the documentation being too confusing, so the goal of this pull request is to make the documentation less confusing, fix its mistakes, and update it.

commit 06705d17750a5872f2205544cf219739f4bf2300:
- Use pronoun-verb contractions to be consistent with the rest of the documentation (This has been done throughout the document)
- Change the name for the `VIEW_CHANNEL` permission on the Discord client
- Change "Set Up the Bot" to "Setting Up the Bot" to be consistent with the rest of the documentation
- Add a missing space
- Make the steps less confusing
- Fix grammar mistakes

commits f5404bfe6fb61b111f086f394993e855fa66064e: and 62e9baab4679cbb307a2448019722f65a05c5f4d
- Trying to make the substeps 1 and 2 clearer

commit 314b1408eabf626761088f7454a3f56c8c0599d3:
- Make the substeps regular steps

commit 6c0fb6e6afd1f0d9a3193224332d508386d6c45a:
- Make the indirect object of the verb "give" clear by replacing "it" with "the bot"

commit b5851cf4fd24d6bcdb59f190d0d3139c0ee98288:
- Fix a typo
- Add information regarding placeholders in commands
- Add a definition for "admin"
- Fix grammar mistakes and punctutation
- Add information regarding the use of message IDs for managing suggestions
- Change "via" to "by running" for clarity
- Replace "a user with the Manage Server (`MANAGE_GUILD`) permission" with "admin"
- Change "staff members" to "other staff members" for clarity
- Change "doing" to "by running" for clarity
- Change "just instead information of the suggestion being rejected will be posted" to "but information of the suggestion being rejected will be posted instead" for clarity
- Remove outdated time expressions and information such as "A newly introduced feature as of September 29, 2018" and "To see the available options as well as the option you have configured, use the command `setvotes`"

commit 7da7cd4095a41d47a7467b1daec81c828a29a034:
- Add spaces

commit 803e5d4b063d0c2430e4e9caa9ee7a86e8acd279:
- Delete the `managing-suggestions-7.png` file because it shows an outdated embed

commit 128f56aea8d183ff7e326cc8226397e9f442fcb1:
- Add a new `managing-suggestions-7.png` file

Ignore the commits e761c98b24f6e5251b7887779db9eb400bbdc215 and `2bbccad7125aee7f9724e0684ac9bb0f0bbeb54f`

commit a11dd9ebf0bfd8ba2c6351d06e4eaede4a24e47f:
- Replace "do" with "run" for clarity (This has been done throughout the document)
- Replace "ex." with "e.g." for clarity
- Remove the outdated time expression "A newly introduced feature as of December 2, 2018" 
- Add missing commas

commit 218be1e259ec47c4722ca0a067baf6e3b832f0f0:
- Add more information on staff suggestions

commit 1cdd379e73ff9bf56786a406fdf9b35d02c7d2d0:
- Trying to make a line break

commit 67c8be667e394f476bd197659405fbb5742b9d91:
- Remove "is something that" as it's unnecessary

commit c07dd8e069963efc2c89007a439c305627ab1fd2:
- Fix punctuation and add the missing word "also"

commit 86d78e4e741b8329f97047cb0d080963f66e6531:
- Change "admin" to "administrator"

commit 094819c437e43880600682fad150ee2d60ea3d87:
- Clarify the "Setting Things Up" subsection and fix punctuation

commit 9cefeb85b5bb253717af35a99ec693f4d40e5708:
- Clarify the "Setting Things Up" subsection

commit 1b99f3806a85fafa1b05ec333162ad3da270210c:
- Make it clear that the configured emoji set doesn't apply to staff suggestions

commit 462966eeec827f58f8c8928132f97776f786ba28:
- Clarify the tip
- Replace "guild" with "server" (This has been done throughout the document)
- Fix grammar mistakes

commit c9bdadcaf28a69f8b414fe06ef72a159906370f9:
- Replace "Suggestions bot" with "bot"
- Fix a typo
- Fix grammar mistakes
- Remove the unnecessary preposition "to" after "posted"
- Replace "will not be allowed to" with "can't" to remove wordiness

commits 348093c7a2e383c59192fd73df87988528367c35, 9aa330ad00f7db7eb608a6b71ba0c63f3849aba2, b14aa92e7f6adb07d860bc0c1d8328f092b360d0, and f781a66a8d64077d05559783ba3f00adbfbf3d5c:
- Trying to make line breaks

commit 945c0270de1e2d7fdcd92f7ad2baa2463251508c:
- Add proper spaces to make line breaks

commit 06b3c3dd8f6f2f9332b2d5aa410ea1c9e9799268:
- Add the symbol `|` to the key
- Fix typos and grammar mistakes
- Remove an useless hyphen
- Add `message ID` to some command syntaxes
- Add the new aliases for `approve` and `reject` commands
- Make the description of the `changelog` command clear
- Update the descriptions of `help`, `info`, and `invite` commands

commit e612f268676b9e768a6509744fcd131e1fee6cc1:
- Add a line break

commit cfa9d10ed133e95fac7797aece4055e2a65829b7:
- Add a line break
- Add the symbol `|` to the key
- Add `message ID` to the command syntaxes

commit 21f8b835fcc86cd9c2ca06e0f0dc57c12d8d8daa:
- Bring back the substeps and make them clearer

commit 96445f340040633a7400d326298d5a920c7e208f:
- Add a line break

commits bcebe90992d91324877939bc19d4175e0f8b6191 and a6bdbda19bd2832565b7a83152c6a7310a01afd0:
- Replace "guild" with "server" in the "Getting Started" section

commit fce1ee750a32865b2f74b05e65e0e6a8b067404f:
- Make the note on placeholders clearer

commit c832b5f033291e4fd38b9d90bbc02736fcc2730a:
- Remove the word "not" afer the contraction "won't"

commit 5ed79c8c7a136ecca2846b062c87bdff31a22a91:
- Remove an unnecessary article

commit 3667436a811ba4201932d1f949abf847fdaf9b75:
- Make the warning clearer
- Change "suggestions role" to "staff role" for clarity
- Bring back the substeps and make them clearer
- Change "`@everyone`" to "the staff role" in substep b) of step 1

commit bfdb57b0e26b9970b95cb6cfe913ad1ecb732496:
- Add a line break in the warning

commit 0977a0e3857a7de207fa5f6f769e78c107968983:
- Replace "ex." with "e.g." for clarity

commit 2a22d752ecffd7eaf002485db95f5740b9338476:
- Put spaces between the steps to make them easier to read

Ignore the commits e9801fa057ce0c36b1a65972d17260e0245705f2 and 08a9d0dab1623e16a76757d73bbccec4d27a8de3

commit 45508ebe910b829c97bd4be50b5708d8fa842f09:
- Put spaces between the steps to make them easier to read

commit 64ad6599166e815da2fec3f80b1ed9d623101f8c:
- Fix the number of the last step

Davi-the-Mudkip committed:
Update README.md

Davi-the-Mudkip committed:
Update README.md

Davi-the-Mudkip committed:
Update README.md

Davi-the-Mudkip committed:
Update README.md

Davi-the-Mudkip committed:
Update README.md

Davi-the-Mudkip committed:
Update managing-suggestions.md

Davi-the-Mudkip committed:
Update managing-suggestions.md

Davi-the-Mudkip committed:
Delete managing-suggestions-7.png

Davi-the-Mudkip committed:
Add files via upload

Davi-the-Mudkip committed:
Update managing-suggestions.md

Davi-the-Mudkip committed:
Update managing-suggestions.md

Davi-the-Mudkip committed:
Update managing-suggestions.md

Davi-the-Mudkip committed:
Update staff-suggestions.md

Davi-the-Mudkip committed:
Update staff-suggestions.md

Davi-the-Mudkip committed:
Update staff-suggestions.md

Davi-the-Mudkip committed:
Fixing punctuation

Davi-the-Mudkip committed:
Change "admin" to "administrator"

Davi-the-Mudkip committed:
Clarify the "Setting Things Up" subsection and fix punctuation

Davi-the-Mudkip committed:
Update README.md

Davi-the-Mudkip committed:
Update staff-suggestions.md

Davi-the-Mudkip committed:
Update statistics.md

Davi-the-Mudkip committed:
Update configuration.md

Davi-the-Mudkip committed:
Update configuration.md

Davi-the-Mudkip committed:
Update configuration.md

Davi-the-Mudkip committed:
Update configuration.md

Davi-the-Mudkip committed:
Update configuration.md

Davi-the-Mudkip committed:
Update configuration.md

Davi-the-Mudkip committed:
Update commands.md

Davi-the-Mudkip committed:
Update commands.md

Davi-the-Mudkip committed:
Update managing-suggestions.md

Davi-the-Mudkip committed:
Update README.md

Davi-the-Mudkip committed:
Update README.md

Davi-the-Mudkip committed:
Update README.md

Davi-the-Mudkip committed:
Update README.md

Davi-the-Mudkip committed:
Update managing-suggestions.md

Davi-the-Mudkip committed:
Update managing-suggestions.md

Davi-the-Mudkip committed:
Update managing-suggestions.md

Davi-the-Mudkip committed:
Update staff-suggestions.md

Davi-the-Mudkip committed:
Update staff-suggestions.md

Davi-the-Mudkip committed:
Update configuration.md

Davi-the-Mudkip committed:
Update commands.md

Davi-the-Mudkip commented:
I think the edits are done and this PR is ready for review.

Davi-the-Mudkip committed:
Update README.md

Davi-the-Mudkip committed:
Update README.md

Davi-the-Mudkip committed:
Update README.md

Davi-the-Mudkip committed:
Update README.md

Davi-the-Mudkip committed:
Update staff-suggestions.md

Davi-the-Mudkip committed:
Update staff-suggestions.md

Skelmis commented on docs/docs/README.md:
```diff
@@ -11,28 +11,37 @@ meta:
+Once the bot is in your server, there are a few housekeeping things you need to do. (You must have the Manage Server (`MANAGE_GUILD`) permission to complete these steps.)
 
-Also by default, the prefix is`,`and the bot searches for `#suggestions` as the default user suggestions channel. The bot mention (`@Suggestions#2602`) can be used as a prefix for commands as well if you ever incorrectly set the prefix or forget the prefix in general.
+Also by default, the prefix is `,` and the bot searches for `#suggestions` as the default suggestions channel. The bot mention (`@Suggestions#2602`) can be used as a prefix for commands as well if you ever incorrectly set the prefix or forget the prefix in general.
```
``and the bot searches for `#suggestions` as the default suggestions channel.`` This is no longer the default behavior as far as I am aware. We should now point users to use their own channel which they must configure.

acollierr17 replied:
The bot will only search for a channel called `#suggestions` when using the `suggest` command. If the channel doesn't exist when the user is running a command, the error will display prompting the channel to be set. Otherwise, the channel will be set before the rest of the `suggest` command is executed. So technically, this message here is still correct. 

I originally removed it but re-added it due to the confusion and issues it was causing. I removed it in the first place unknowing because I didn't know that the `messageReactionAdd` event was broken (https://github.com/suggestionsbot/suggestions-bot/commit/bb0decf96fb899c97d7c8eb59c6a52cff6139df4)

Skelmis commented on docs/docs/commands.md:
```diff
@@ -12,24 +12,25 @@ meta:
 | info | View bot information      |    [click](#info-command) |
 | invite | Receive a DM with information on inviting the bot to your server | [click](#invite-command) | 
-| patrons | View all Patrons who have pledged to Nerd Cave Development | [click](#patrons-command) |
+| patrons | View all Patrons who've pledged to Nerd Cave Development | [click](#patrons-command) |
```
This command no longer exists to the best of my knowledge.

Davi-the-Mudkip replied:
Thanks for this information

Skelmis commented on docs/docs/commands.md:
````diff
@@ -89,37 +92,38 @@ This command sends a message of the current suggestions channel. If no suggestio
 ,patrons
 ```
-View a list of all current Patrons who support Nerd Cave Development. These people, pledges big or small, are much appreciated and lets the developers know good is being done.
+View a list of all current Patrons who support Nerd Cave Development. These people, pledge big or small, are much appreciated and let the developers know good is being done.
````
This command has been removed as far as I'm aware.

acollierr17 replied:
This is correct

Skelmis reviewed (changes requested):
@acollierr17 can just confirm my thoughts towards my review points.

Otherwise looking really good.

acollierr17 commented on docs/docs/commands.md:
```diff
@@ -40,44 +41,46 @@ These commands require a staff role or the Manage Server (`MANAGE_GUILD`) permis
 | approve | Approve a submitted suggestion | [click](#approve-command) |
 | note | Add a note to a submitted suggestion | [click](#note-command) |
 | reject | Reject a submitted suggestion | [click](#reject-command) | 
-| staffsuggest | Submit a staff suggestion to be viewed by staff-only | [click](#staff-suggest-command) |
```
That hyphen placement you removed was appropriate.

acollierr17 replied:
Whoops, I didn't clarify this correctly. Basically, it should still be `staff-only` and not `staff only` as that hyphen I originally had was an appropriate use. Other than that, I think this is all set unless @Skelmis has anything else he'd like to add on?

acollierr17 commented on docs/docs/commands.md:
````diff
@@ -89,37 +92,38 @@ This command sends a message of the current suggestions channel. If no suggestio
 ,invite
 ```
-The invite command sends the user a DM with the bot invite link as well as the link to this documenation and the [Support Discord](https://discord.gg/ntXkRan).
+The invite command sends the user a DM with the bot invite link as well aslinks to the bot website, the [Support Discord](https://discord.gg/ntXkRan), and the GitHub page.
````
I would add a hyperlink here to the [GitHub](https://github.com/suggestionsbot/suggestions-bot): `[GitHub](https://github.com/suggestionsbot/suggestions-bot)`

acollierr17 commented on docs/docs/configuration.md:
```diff
@@ -30,66 +34,66 @@ Here is an overview of all the configurable options. Keep scrolling for more inf
 Usage: `config staffchannel` or `config staffchannel [channel]`
 
-You can either view or set the channel where staff suggestions will be posted to. The options to pass through follow the same as setting the [Suggestions Channel](#suggestions-channel). Keep in mind you cannot set a channel that does not exist.
+You can either view or set the channel where staff suggestions will be posted. The options to pass through follow the same as setting the [Suggestions Channel](#suggestions-channel). Keep in mind you can't set a channel that doesn't exist.
```
I would even extend this by also saying you can't set a channel that doesn't exist _in the server_.

Davi-the-Mudkip replied:
Done

acollierr17 reviewed (changes requested):
Looks solid to me aside from a couple of things I'd like to further review/change.

Davi-the-Mudkip committed:
Remove the non-existent patrons command

Davi-the-Mudkip committed:
Add hyperlinks to the bot's GitHub page

Davi-the-Mudkip committed:
Adding the phrase "in the server" for clarification

I also replaced "Invalid channels" and "Invalid roles" with "Channels that doesn't exist in the server" and "Roles that doesn't exist in the server" respectively.

Davi-the-Mudkip committed:
Readd the hyphen

acollierr17 commented on docs/docs/commands.md:
````diff
@@ -89,37 +91,32 @@ This command sends a message of the current suggestions channel. If no suggestio
 ```
-The help command displays all commands (commands will display based on permissions of the user) and additonal information dislaying current prefix, suggestions channel, bug reports links and the link to this documentation.
+The help command displays all commands (commands will display based on permissions of the user) and additonal information: current prefix, suggestions channel, a link to the website, a link to the GitHub page, and a link to the support server.  
+If you type a command aliase after `help`, the bot will display help information on the relevant command.
````
Should be spelled `alias`.

acollierr17 commented on docs/docs/configuration.md:
```diff
@@ -30,66 +34,66 @@ Here is an overview of all the configurable options. Keep scrolling for more inf
 Usage: `config channel` or `config channel [channel]`
 
-You can either view or set the channel where suggestions will be posted to. Invalid channels will not be allowed to be set. The options to pass through when setting the channel can be:
+You can either view or set the channel where suggestions will be posted. Channels that doesn't exist in the server can't be set. The options to pass through when setting the channel can be:
```
Should be `Channels that don't exist` for proper verb usage.

acollierr17 commented on docs/docs/configuration.md:
```diff
@@ -30,66 +34,66 @@ Here is an overview of all the configurable options. Keep scrolling for more inf
 Usage: `config roles` or `config roles [role]`
 
-You can either view or set the various staff roles that are used for managing suggestions as well as submitting staff suggestions. Invalid roles will not be allowed to be set. The options to pass through when adding/removing a staff role can be:
+You can either view or set the various staff roles that are used for managing suggestions as well as submitting staff suggestions. Roles that doesn't exist in the server can't be set. The options to pass through when adding/removing a staff role can be:
```
Should be `Roles that don't exist exist` for proper verb usage.

acollierr17 commented on docs/docs/managing-suggestions.md:
```diff
@@ -13,45 +13,51 @@ _**Key:**_
 
 ![Suggestion Approved DM](/images/managing-suggestions-4.png)
 
-It seems like a lot doesn't it? It really isn't because it all happens in the matter of seconds!
```
Honestly, it can probably be said as "milliseconds" instead of "seconds." If an act like this is taking a couple of seconds, I'd be concerned.

acollierr17 reviewed (changes requested):
A couple of grammar things to take care of and I think we're just about set.

Davi-the-Mudkip committed:
Fix a mistake

Davi-the-Mudkip committed:
Fix other mistakes
